### PR TITLE
meta: omit LD_LIBRARY_PATH and PATH in classic confinement

### PR DESCRIPTION
**Pull request by mr-cal**
- [X] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [X] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [X] Have you successfully run `make lint`?
- [X] Have you successfully run `pytest tests/unit`?

-----

When confinement is classic, do not use the default values for `LD_LIBRARY_PATH` and `PATH`.

Includes an optimization/refactor suggestion from my original implementation: https://github.com/snapcore/snapcraft/pull/3758#discussion_r881900290

Resolves https://github.com/snapcore/snapcraft/issues/4187

(CRAFT-1774)

### CODE REVIEW

File: tests/unit/meta/test_snap_yaml.py [hunk 1]
```diff
@@ -593,6 +593,62 @@ def test_project_environment_ld_library_path_null(simple_project, new_dir):
...
 
+@pytest.mark.parametrize(
+    "ld_library_path",
+    [{}, {"LD_LIBRARY_PATH": None}, {"LD_LIBRARY_PATH": "test-ld-library-path"}],
+)
+@pytest.mark.parametrize("path", [{}, {"PATH": None}, {"PATH": "test-path"}])
+@pytest.mark.parametrize(
+    "other_var", [{}, {"OTHER_VAR": None}, {"OTHER_VAR": "test-foo"}]
+)
+def test_project_environment_classic_confinement(
```
Comment (cmatsuoka) [inline tests/unit/meta/test_snap_yaml.py:604]: Are we testing non-classic to make sure it's still working?

File: tests/unit/meta/test_snap_yaml.py [hunk 2]
```diff
@@ -593,6 +593,62 @@ def test_project_environment_ld_library_path_null(simple_project, new_dir):
...
+        prime_dir=Path(new_dir),
+        arch="amd64",
+    )
+    yaml_file = Path("meta/snap.yaml")
+    assert yaml_file.is_file()
+
+    content = yaml_file.read_text()
+    assert (
+        content
+        == textwrap.dedent(
```
[the remark this answers is not in the document]
Comment (mr-cal) [reply tests/unit/meta/test_snap_yaml.py:632]: I agree, but black keeps changing it to this style.

File: tests/unit/meta/test_snap_yaml.py (hunk 1 above)
Comment (mr-cal) [reply tests/unit/meta/test_snap_yaml.py:604]: Yes, the existing tests use `strict` confinement.

Comment (mr-cal) [reply tests/unit/meta/test_snap_yaml.py:604]: I started to refactor and parameterize the existing tests but decided not to change them to ensure I am not changing non-classic behavior.

**Comment by AlanGriffiths** [comment]
I realise this has merged, but this seems the best place to call out problems. Many of the classic snaps I maintain rely on having executables on `$PATH` (typically `$SNAP/bin` or `$SNAP/usr/bin`).

I doubt that I'm the only one using the existing behaviour of adding these to `$PATH`. While I have sympathy for the change dropping them (my snaps need logic to strip them) this is probably going to break a lot of snaps if promoted in the current form.

**Pull request merged**